### PR TITLE
Add new client for Etherscan API v2

### PR DESCRIPTION
**Pull request by falvaradorodriguez**
- Closes: #1398

More info about the migration: https://docs.etherscan.io/etherscan-v2/getting-started/v2-quickstart

### CODE REVIEW

File: safe_eth/eth/clients/etherscan_client_v2.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        request_timeout: int = int(
+            os.environ.get("ETHERSCAN_CLIENT_REQUEST_TIMEOUT", 10)
+        ),
+    ):
+        super().__init__(EthereumNetwork.MAINNET, api_key, request_timeout)
+        self.network = network
+        self.base_api_url = self.BASE_API_V2_URL
+
+    @classmethod
+    def _get_supported_networks(cls) -> List[Dict[str, Any]]:
```
Comment (Uxio0) [inline safe_eth/eth/clients/etherscan_client_v2.py:27]: I think this should be cached, and also it shouldn't be a private method

File: safe_eth/eth/clients/etherscan_client_v2.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        :param network: The Ethereum network to check.
+        :return: True if the network is supported; False otherwise.
+        """
+        supported_networks = cls._get_supported_networks()
+        return any(
+            item.get("chainid") == str(network.value) for item in supported_networks
+        )
+
+    def build_url(self, path: str) -> str:
```
Comment (Uxio0) [inline safe_eth/eth/clients/etherscan_client_v2.py:52]: This method should be at the top of the class, just for clarity

**Review by Uxio0 — CHANGES_REQUESTED** [review]

File: safe_eth/eth/clients/etherscan_client.py [hunk 3]
```diff
@@ -320,7 +320,7 @@ def __init__(
         self.request_timeout = request_timeout
 
     def build_url(self, path: str):
-        url = urljoin(self.base_api_url, path)
+        url = urljoin(self.base_api_url, f"api?{path}")
```
Comment (moisses89) [inline safe_eth/eth/clients/etherscan_client.py:323]: Why api is out of the path?

File: safe_eth/eth/clients/etherscan_client_v2.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        url = urljoin(self.base_api_url, f"v2/api?chainid={self.network.value}&{path}")
```
Comment (moisses89) [inline safe_eth/eth/clients/etherscan_client_v2.py:53]: `v2/api` should be path?

**Review by moisses89 — APPROVED** [review]

File: safe_eth/eth/clients/etherscan_client_v2.py (hunk 2 above)
Comment (falvaradorodriguez) [reply safe_eth/eth/clients/etherscan_client_v2.py]: Done [here](https://github.com/safe-global/safe-eth-py/pull/1407/commits/9b8049300f80f04aa75dd4b1f643d0735ab3d0ed)

File: safe_eth/eth/clients/etherscan_client_v2.py (hunk 1 above)
Comment (falvaradorodriguez) [reply safe_eth/eth/clients/etherscan_client_v2.py]: IMO this method will be called a few times and may even add a string along the way. It should have no problems without being cached.

It is updated as public method [here](https://github.com/safe-global/safe-eth-py/pull/1407/commits/9b8049300f80f04aa75dd4b1f643d0735ab3d0ed)

File: safe_eth/eth/clients/etherscan_client.py (hunk 3 above)
Comment (falvaradorodriguez) [reply safe_eth/eth/clients/etherscan_client.py]: Maybe the problem is the name of the parameter. I updated it as `query` [here](https://github.com/safe-global/safe-eth-py/pull/1407/commits/9b8049300f80f04aa75dd4b1f643d0735ab3d0ed)

File: safe_eth/eth/clients/etherscan_client_v2.py (hunk 4 above)
Comment (falvaradorodriguez) [reply safe_eth/eth/clients/etherscan_client_v2.py]: Yes, it's the path. I already updated the name of the parameter.

**Review by Uxio0 — APPROVED** [review]

**Pull request merged**